e2e/test/ASCforIoTLogAnalyticsClient.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    public class ASCforIoTLogAnalyticsClient : IDisposable
+    {
+        //Azure Active Directory authentication authority for public cloud
+        private const string AuthenticationAuthorityTemplate = "https://login.windows.net/{0}";
+        //Azure Log Analytics Authentication token audience
+        private const string Audience =  "https://api.loganalytics.io/";
+        //Azure Log Analytics query URL
+        private const string QueryUriTemplate = "https://api.loganalytics.io/{0}/workspaces/{1}/query";
+        //Azure Log Analytics API version
+        private const string LogAnalyticsApiVersion = "v1";
+        //Query template for querying a SecurityIoTRawEvent by device id and raw event id
+        private const string RawEventQueryTemplate =
+            @"SecurityIoTRawEvent
+    | where DeviceId == ""{0}""
+    | where IoTRawEventId == ""{1}""";
+        
+        private readonly string _workspaceId = Configuration.ASCforIoTLogAnalytics.WorkspacedId;
+        private readonly string _aadTenant = Configuration.ASCforIoTLogAnalytics.AadTenant;
+        private readonly string _appId = Configuration.ASCforIoTLogAnalytics.AadAppId;
+        private readonly string _appCertificate = Configuration.ASCforIoTLogAnalytics.AadAppCertificate;
+
+        private readonly TimeSpan _polingInterval = TimeSpan.FromSeconds(20);
+
+        private readonly HttpClient _client;
```

Comment:
Please use the client - none of the new features are supported on NET451. Use conditional compilation in the e2e tests to avoid compiling only for netstandard2.0 / netcoreapp2.1.